js/interface.js

```diff
@@ -96,11 +96,53 @@ function parseThumbnail(file) {
   file.thumbnail = Fliplet.Media.authenticate(file.url.replace(Fliplet.Env.get('apiUrl'), Fliplet.Env.get('apiCdnUrl')));
 }
 
-function navigateToFolder(item) {
-  if (!item.data('app-id')) {
-    $('[data-org-id="' + item.data('org-id') +'"][data-browse-folder]').click();
+function navigateToRootLevelFolder(options) {
+  var $itemFolder = options.appId
+    ? $('[data-app-id="' + options.appId + '"][data-browse-folder]')
+    : $('[data-org-id="' + options.orgId +'"][data-browse-folder]');
+
+  // Deselection of  all active files when user switches folder
+  $itemFolder.parents('.file-manager-body').find('.file-row.active input[type="checkbox"]').click();
+  $('[restore-action], [file-remove-trash]').hide();
+  $('[rename-action], [delete-action]').show();
+
+  disableSearchState();
+  resetUpTo($itemFolder);
+  updateSearchTypeOptions($itemFolder.data('type'));
+
+  return getFolderContents($itemFolder, true);
+}
+
+function navigateToFolder($item) {
+  var id = $item.data('id');
+  var backItem;
+
+  removeSelection();
+  hideSideActions();
+  disableSearchState();
+
+  // Making a backstack item
+  backItem = _.find(folders, ['id', id]);
+  backItem.tempElement = $item;
+  backItem.back = function() {
+    getFolderContents(backItem.tempElement);
+  };
+  backItem.type = 'folderId';
+
+  navStack.push(backItem);
+
+  updatePaths();
+  getFolderContents($item);
+}
+
+function navigateToFolderItem(item) {
+  if (item.data('folder')) {
+    navigateToRootLevelFolder({ appId: item.data('app-id'), orgId: item.data('org-id') })
```

Comment:
I'd rather you only pass `appId` or `orgId`, depending on how you want the `navigateToRootFolder` function to behave.

The `navigateToRootFolder` function can also check to make sure that it prioritises `appId`. This way, the caller is clear in its intentions and the function is able to handle the scenarios too.

Reply:
@tonytlwu I've added a suggestion and I hope I understand you correctly with that suggestion.